rust/blockstore/src/arrow/block/delta/ordered_merge_delta.rs

```diff
@@ -0,0 +1,227 @@
+use std::collections::HashMap;
+
+use super::{storage::BlockStorage, types::DeltaCommon};
+use crate::{
+    arrow::{
+        block::Block,
+        types::{ArrowReadableKey, ArrowReadableValue, ArrowWriteableKey, ArrowWriteableValue},
+    },
+    key::{CompositeKey, KeyWrapper},
+};
+use arrow::array::{RecordBatch, StringArray};
+use uuid::Uuid;
+
+/// A block delta tracks a source block and represents the new state of a block. Blocks are
+/// immutable, so when a write is made to a block, a new block is created with the new state.
+/// A block delta is a temporary representation of the new state of a block. A block delta
+/// can be converted to a block data, which is then used to create a new block. A block data
+/// can be converted into a block delta for new writes.
+/// # Methods
+/// - add: adds a key value pair to the block delta.
+/// - delete: deletes a key from the block delta.
+/// - get_size: gets the size of the block delta.
+/// - split: splits the block delta into new block deltas based on a max block size.
+pub struct OrderedMergeBlockDelta {
```

Comment:
don't love this name, open to alternatives

but will also be renaming delta in the near future to builder